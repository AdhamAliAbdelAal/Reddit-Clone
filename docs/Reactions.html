<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Reactions</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Reactions</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Reactions<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Reactions"><span class="type-signature"></span>new Reactions<span class="signature">()</span><span class="type-signature"> &rarr; {React.Component}</span></h4>
    

    



<div class="description">
    <p>This component is the container of reactions</p>
</div>













    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>flexDirection</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>The direction of reactions (vertical or horizontal).</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>votes</code></td>
            

            <td class="type">
            
                
<span class="param-type">number</span>


            
            </td>

            

            

            <td class="description last"><p>Number of post votes.</p></td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_Post_Reactions_Reactions.jsx.html">components/Post/Reactions/Reactions.jsx</a>, <a href="components_Post_Reactions_Reactions.jsx.html#line15">line 15</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Upvote and downvote only.</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">React.Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="About.html">About</a></li><li><a href="AddList.html">AddList</a></li><li><a href="AddSector.html">AddSector</a></li><li><a href="BackHomeBottun.html">BackHomeBottun</a></li><li><a href="BackToTop.html">BackToTop</a></li><li><a href="Comments.html">Comments</a></li><li><a href="CommentsContent.html">CommentsContent</a></li><li><a href="CommentsHeader.html">CommentsHeader</a></li><li><a href="Communities.html">Communities</a></li><li><a href="CommunityItem.html">CommunityItem</a></li><li><a href="CommunityList.html">CommunityList</a></li><li><a href="Content.html">Content</a></li><li><a href="CreatePostContainer.html">CreatePostContainer</a></li><li><a href="CreatePostForm.html">CreatePostForm</a></li><li><a href="CreatePostInHome.html">CreatePostInHome</a></li><li><a href="DrawerBottom.html">DrawerBottom</a></li><li><a href="Email.html">Email</a></li><li><a href="EmptyContent.html">EmptyContent</a></li><li><a href="Filter.html">Filter</a></li><li><a href="FilterFull.html">FilterFull</a></li><li><a href="FilterSmall.html">FilterSmall</a></li><li><a href="FirstParty.html">FirstParty</a></li><li><a href="Flirt.html">Flirt</a></li><li><a href="ForgetPassword.html">ForgetPassword</a></li><li><a href="ForgetUsername.html">ForgetUsername</a></li><li><a href="FormDialog.html">FormDialog</a></li><li><a href="Header.html">Header</a></li><li><a href="HomeList.html">HomeList</a></li><li><a href="HomePageContainer.html">HomePageContainer</a></li><li><a href="HomePageFooter.html">HomePageFooter</a></li><li><a href="Input.html">Input</a></li><li><a href="JoinButton.html">JoinButton</a></li><li><a href="LoadingPage.html">LoadingPage</a></li><li><a href="global.html#LogIn">LogIn</a></li><li><a href="LogInPage.html">LogInPage</a></li><li><a href="LogoIcon.html">LogoIcon</a></li><li><a href="MainContent.html">MainContent</a></li><li><a href="Moderation.html">Moderation</a></li><li><a href="ModeratorList.html">ModeratorList</a></li><li><a href="Moderators.html">Moderators</a></li><li><a href="NavIcons.html">NavIcons</a></li><li><a href="Navbar.html">Navbar</a></li><li><a href="NotificationCategories.html">NotificationCategories</a></li><li><a href="NotificationImages.html">NotificationImages</a></li><li><a href="Notifications.html">Notifications</a></li><li><a href="NotificationsBody.html">NotificationsBody</a></li><li><a href="NotificationsHeader.html">NotificationsHeader</a></li><li><a href="OtherProfileComments.html">OtherProfileComments</a></li><li><a href="OtherProfileCommentsContent.html">OtherProfileCommentsContent</a></li><li><a href="OtherProfileCommentsHeader.html">OtherProfileCommentsHeader</a></li><li><a href="OtherProfileCommunities.html">OtherProfileCommunities</a></li><li><a href="OtherProfileContent.html">OtherProfileContent</a></li><li><a href="OtherProfileEmptyContent.html">OtherProfileEmptyContent</a></li><li><a href="OtherProfileFilter.html">OtherProfileFilter</a></li><li><a href="OtherProfileFilterFull.html">OtherProfileFilterFull</a></li><li><a href="OtherProfileFilterSmall.html">OtherProfileFilterSmall</a></li><li><a href="OtherProfileHeader.html">OtherProfileHeader</a></li><li><a href="OtherProfileMainContent.html">OtherProfileMainContent</a></li><li><a href="OtherProfilePost.html">OtherProfilePost</a></li><li><a href="OtherProfilePostFooter.html">OtherProfilePostFooter</a></li><li><a href="OtherProfilePostFooterListResponsive.html">OtherProfilePostFooterListResponsive</a></li><li><a href="OtherProfilePostHeader.html">OtherProfilePostHeader</a></li><li><a href="OtherProfilePostSide.html">OtherProfilePostSide</a></li><li><a href="OtherProfilePosts.html">OtherProfilePosts</a></li><li><a href="OtherProfilePostsTap.html">OtherProfilePostsTap</a></li><li><a href="OtherProfileSidebar.html">OtherProfileSidebar</a></li><li><a href="OtherProfileUserInfo.html">OtherProfileUserInfo</a></li><li><a href="PersonalReddit.html">PersonalReddit</a></li><li><a href="Post.html">Post</a></li><li><a href="PostFooter.html">PostFooter</a></li><li><a href="PostFooterList.html">PostFooterList</a></li><li><a href="PostFooterListResponsive.html">PostFooterListResponsive</a></li><li><a href="PostHeader.html">PostHeader</a></li><li><a href="PostList.html">PostList</a></li><li><a href="PostReactions.html">PostReactions</a></li><li><a href="PostSide.html">PostSide</a></li><li><a href="PostSubmission.html">PostSubmission</a></li><li><a href="PostTags.html">PostTags</a></li><li><a href="PostTypes.html">PostTypes</a></li><li><a href="Posts.html">Posts</a></li><li><a href="PostsClassification.html">PostsClassification</a></li><li><a href="PostsFilteredTap.html">PostsFilteredTap</a></li><li><a href="PostsTap.html">PostsTap</a></li><li><a href="ProfileHeader.html">ProfileHeader</a></li><li><a href="ProfileMainContent.html">ProfileMainContent</a></li><li><a href="RadioBtn.html">RadioBtn</a></li><li><a href="Reactions.html">Reactions</a></li><li><a href="RedditPremium.html">RedditPremium</a></li><li><a href="SNavbar.html">SNavbar</a></li><li><a href="SearchField.html">SearchField</a></li><li><a href="SideBar.html">SideBar</a></li><li><a href="Sidebar_.html">Sidebar</a></li><li><a href="SignUp.html">SignUp</a></li><li><a href="global.html#SignUpEmail">SignUpEmail</a></li><li><a href="SignUpUsername.html">SignUpUsername</a></li><li><a href="SubredditsMenu.html">SubredditsMenu</a></li><li><a href="ThirdParty.html">ThirdParty</a></li><li><a href="global.html#ThirdPartyButton">ThirdPartyButton</a></li><li><a href="UserInfo.html">UserInfo</a></li><li><a href="UserList.html">UserList</a></li><li><a href="UserNamePasswordForm.html">UserNamePasswordForm</a></li><li><a href="module.html#.exports">exports</a></li></ul><h3>Global</h3><ul><li><a href="global.html#ThirdPartyContainer">ThirdPartyContainer</a></li><li><a href="global.html#checkPassword">checkPassword</a></li><li><a href="global.html#checkUserName">checkUserName</a></li><li><a href="global.html#matchPassword">matchPassword</a></li><li><a href="global.html#refreshUsernames">refreshUsernames</a></li><li><a href="global.html#responseFacebook">responseFacebook</a></li><li><a href="global.html#responseGoogleSuccess">responseGoogleSuccess</a></li><li><a href="global.html#signUp">signUp</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.0</a> on Sat Nov 12 2022 22:51:45 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>