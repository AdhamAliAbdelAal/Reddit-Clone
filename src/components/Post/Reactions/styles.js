import { Box, IconButton } from '@mui/material';
import { styled } from '@mui/system';

export const Voting = styled(Box)(({ flexDirection, viewpost }) => ({
  padding: (flexDirection === 'column' ? '8px 4px' : 0),
  backgroundColor: !viewpost ? (flexDirection === 'column' ? '#F8F9FA' : '#fff') : '#fff',
  display: 'flex',
  borderRadius: '3px 0 0 3px',
  alignItems: 'center',
}));

export const Votes = styled(Box)(() => ({
  fontSize: '12px',
  fontWeight: 'bolder',
  cursor: 'pointer',
}));

export const ReactionIconButton = styled(IconButton)(() => ({
  borderRadius: 4,
  padding: '2px 3px',
}));
