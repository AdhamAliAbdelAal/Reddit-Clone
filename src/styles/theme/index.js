import { createTheme } from '@mui/material';

const theme = createTheme({
  palette: {
    primary: {
      main: '#0079D3',
      light: '#1484D6',
      dark: '#1484D6',
      fade: '#3394DC',
    },
    secondary: {
      main: '#FF4500',
      light: '#FF5414',
      dark: '#FF5414',
    },
    third: {
      main: '#898989',
      light: '#E8E8E8',
      dark: '#E8E8E8',
    },
    action: {
      hoverOpacity: 0.1,
    },
    neutral: {
      main: '#978A8C',
      contrastText: '#fff',
    },
    white: {
      main: '#ffffff',
      light: '#ffffff',
      dark: '#898989',
    },
    offwhite: {
      main: '#DAE0E6',
      dark: '#2E2F30',
    },
  },
});

export const fonts = {
  'system-ui': 'system-ui',
};
export default theme;
