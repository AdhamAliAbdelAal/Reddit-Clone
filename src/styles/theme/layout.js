import { createTheme } from '@mui/material';

const layoutTheme = (theme) => createTheme({
  ...theme,
  breakpoints: {
    values: {
      xxs: 335,
      xs: 420,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
});
export default layoutTheme;
