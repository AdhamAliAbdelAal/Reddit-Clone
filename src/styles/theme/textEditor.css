.rdw-option-wrapper {
    background-color: #f6f7f8 !important;
    padding: 5px 3px !important;
    border-radius: 7px !important;
    border-width: 0 !important;
    margin: 0 3px !important;
  }
  .rdw-option-wrapper:hover,.rdw-option-active{
    background-color: #1a1a1b1a !important;
    box-shadow: none !important;
  }

.public-DraftStyleDefault-block{
    margin: 0 !important;
    margin-top: 5px !important;
}