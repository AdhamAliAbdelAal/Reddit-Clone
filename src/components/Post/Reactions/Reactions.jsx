import { useTheme } from '@mui/system';
import { useReducer } from 'react';
import { ReactionIconButton, Votes, Voting } from './styles';
import VoteIcon from '../../VoteIcons/VoteIcon/VoteIcon';
import VotedIcon from '../../VoteIcons/VotedIcon/VotedIcon';
/**
 * This component is the container of reactions
 *
 * @component Reactions
 * @property {string} flexDirection -The direction of reactions (vertical or horizontal).
 * @property {number} votes -Number of post votes.
 * @returns {React.Component}  Upvote and downvote only.
 */

function Reactions(props) {
  const { flexDirection, votes } = props;
  const theme = useTheme();
  const reducer = (state, action) => {
    switch (action) {
      case 'upvote':
        return 1;
      case 'downvote':
        return -1;
      default:
        return 0;
    }
  };
  const [reaction, dispatch] = useReducer(reducer, 0);
  return (
    <Voting
      flexDirection={flexDirection}
      gap={0.5}
    >
      {(reaction === 1)
        ? (
          <ReactionIconButton
            onClick={() => dispatch('cancel upvote')}
            data-testid="upvoted button"
          >
            <VotedIcon
              color={theme.palette.secondary?.main}
              direction="up"
            />
          </ReactionIconButton>
        )
        : (
          <ReactionIconButton
            onClick={() => dispatch('upvote')}
            data-testid="upvote button"
          >
            <VoteIcon
              color={theme.palette.secondary?.main}
              direction="up"
            />
          </ReactionIconButton>
        )}
      <Votes color={(reaction === 0 ? '#000' : (reaction === 1 ? theme.palette.secondary?.main : theme.palette.primary?.main))}>
        {votes + reaction}
      </Votes>
      {(reaction === -1)
        ? (
          <ReactionIconButton
            onClick={() => dispatch('cancel downvote')}
            data-testid="downvoted button"
          >
            <VotedIcon
              color={theme.palette.primary?.main}
              direction="down"
            />
          </ReactionIconButton>
        )
        : (
          <ReactionIconButton
            onClick={() => dispatch('downvote')}
            data-testid="downvote button"
          >
            <VoteIcon
              color={theme.palette.primary?.main}
              direction="down"
              data-testid="downvote"
            />
          </ReactionIconButton>
        )}
    </Voting>
  );
}

export default Reactions;
