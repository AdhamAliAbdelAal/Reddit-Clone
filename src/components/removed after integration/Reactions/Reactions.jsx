import ThumbUpOutlinedIcon from '@mui/icons-material/ThumbUpOutlined';
import ThumbUpIcon from '@mui/icons-material/ThumbUp';
import ThumbDownOutlinedIcon from '@mui/icons-material/ThumbDownOutlined';
import ThumbDownIcon from '@mui/icons-material/ThumbDown';
import { useTheme } from '@mui/system';
import { useReducer } from 'react';
import { ReactionIconButton, Votes, Voting } from './styles';

function Reactions({ flexDirection }) {
  const theme = useTheme();
  const reducer = (state, action) => {
    switch (action) {
      case 'upvote':
        return 1;
      case 'downvote':
        return -1;
      default:
        return 0;
    }
  };
  const [reaction, dispatch] = useReducer(reducer, 0);
  return (
    <Voting flexDirection={flexDirection} gap={0.5}>
      {(reaction === 1)
        ? (
          <ReactionIconButton onClick={() => dispatch('cancel upvote')}>
            <ThumbUpIcon
              sx={{
                fontSize: 22,
                color: theme.palette.secondary.main,
              }}
            />
          </ReactionIconButton>
        )
        : (
          <ReactionIconButton onClick={() => dispatch('upvote')}>
            <ThumbUpOutlinedIcon
              sx={{
                fontSize: 22,
                '&:hover': {
                  color: theme.palette.secondary.main,
                },
              }}
            />
          </ReactionIconButton>
        )}
      <Votes color={(reaction === 0 ? '#000' : (reaction === 1 ? theme.palette.secondary.main : theme.palette.primary.main))}>
        {156 + reaction}
      </Votes>
      {(reaction === -1)
        ? (
          <ReactionIconButton onClick={() => dispatch('cancel downvote')}>
            <ThumbDownIcon
              sx={{
                fontSize: 22,
                color: theme.palette.primary.main,
              }}
            />
          </ReactionIconButton>
        )
        : (
          <ReactionIconButton onClick={() => dispatch('downvote')}>
            <ThumbDownOutlinedIcon sx={{
              fontSize: 22,
              '&:hover': {
                color: theme.palette.primary.main,
              },
            }}
            />
          </ReactionIconButton>
        )}
    </Voting>
  );
}

export default Reactions;
