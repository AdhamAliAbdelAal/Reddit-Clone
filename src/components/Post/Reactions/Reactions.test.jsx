import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { ThemeProvider } from '@mui/system';
import { screen, render, fireEvent } from '@testing-library/react';
import theme from '../../../styles/theme';
import Reactions from './Reactions';

describe(Reactions, () => {
  it('Reactions renders correctly', () => {
    const testCase = {
      flexDirection: 'row', votes: 100,
    };
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Reactions
            votes={testCase.votes}
            flexDirection={testCase.flexDirection}
          />
        </ThemeProvider>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Upvoting and downvoting work correctly', () => {
    const testCase = {
      flexDirection: 'row', votes: 100,
    };
    render(
      <ThemeProvider theme={theme}>
        <Reactions
          votes={testCase.votes}
          flexDirection={testCase.flexDirection}
        />
      </ThemeProvider>,
    );
    expect(screen.queryByTestId('upvote button')).toBeInTheDocument();
    expect(screen.queryByTestId('upvoted button')).not.toBeInTheDocument();
    expect(screen.queryByTestId('downvote button')).toBeInTheDocument();
    expect(screen.queryByTestId('downvoted button')).not.toBeInTheDocument();

    fireEvent.click(screen.queryByTestId('upvote button'));

    expect(screen.queryByTestId('upvote button')).not.toBeInTheDocument();
    expect(screen.queryByTestId('upvoted button')).toBeInTheDocument();
    expect(screen.queryByTestId('downvote button')).toBeInTheDocument();
    expect(screen.queryByTestId('downvoted button')).not.toBeInTheDocument();

    fireEvent.click(screen.queryByTestId('downvote button'));

    expect(screen.queryByTestId('upvote button')).toBeInTheDocument();
    expect(screen.queryByTestId('upvoted button')).not.toBeInTheDocument();
    expect(screen.queryByTestId('downvote button')).not.toBeInTheDocument();
    expect(screen.queryByTestId('downvoted button')).toBeInTheDocument();
  });
});
