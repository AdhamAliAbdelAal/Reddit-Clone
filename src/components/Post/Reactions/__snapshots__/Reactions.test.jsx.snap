// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reactions Reactions renders correctly 1`] = `
<div
  className="MuiBox-root css-vgs5br"
>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-xv33dv-MuiButtonBase-root-MuiIconButton-root"
    data-testid="upvote button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1w87cc9-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </button>
  <div
    className="MuiBox-root css-1py4uq7"
  >
    100
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-xv33dv-MuiButtonBase-root-MuiIconButton-root"
    data-testid="downvote button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10d9xhr-MuiSvgIcon-root"
      data-testid="ForwardOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M14 8.83 17.17 12 14 15.17V14H6v-4h8V8.83M12 4v4H4v8h8v4l8-8-8-8z"
      />
    </svg>
  </button>
</div>
`;
